add support for eslint, clean up directory

bentoncase opened the pull request:
I restructured the files so that it will be much easier to use browserify and npm to pull in third party packages in the future. One of my upcoming pull requests will do this.

I also added support for eslint, and it looks like there are about 500 errors in content.js. @groovecoder what should we do about this? Fix linting errors in a separate pull request?

Commit:
add support for eslint

groovecoder reviewed (changes requested):
Hmm ... since this code is copied from OpenWPM and Privacy Badger, I don't know that we want to enforce those cleanups. Probably just skip the `content.js` file unless/until we start changing its contents ourselves.

We also need to update the "Run it" section of the README to be `web-ext run -s src/`

Commit:
eslint ignore content.js, update readme

casebenton commented:
@groovecoder how does it look now? I removed the readme comment about web-ext since it's now a dev dependency

groovecoder commented on README.md:
```diff
@@ -14,13 +14,10 @@ anything in a database. If you need all that good stuff, you should really look
 cd data-leak
-web-ext run
+npm install
+npm run build
```
This builds a `.zip` but doesn't actually run the browser with the add-on. Let's keep it `web-ext run -s src`

Commit:
Update README.md

is it alright if we leave the npm run build step in? (For bundling with browserify, not added yet but will be soon)

casebenton commented:
@groovecoder r?

groovecoder reviewed (approved):
Technically the build step isn't necessary, but this works.
